exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1443 @@
+	dates, err := kline.CalculateCandleDateRanges(start, end, interval, by.Features.Enabled.Kline.ResultLimit)
+	if err != nil {
+		return kline.Item{}, err
+	}
+
+	switch a {
+	case asset.Spot:
+		candles, err := by.GetKlines(ctx, pair.String(), by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start, end)
+		if err != nil {
+			return kline.Item{}, err
+		}
+
+		for i := range candles {
+			for j := range ret.Candles {
+				if ret.Candles[j].Time.Equal(time.Unix(int64(candles[i].Open), 0)) {
+					continue
+				}
+			}
+			ret.Candles = append(ret.Candles, kline.Candle{
+				Time:   time.Unix(int64(candles[i].Open), 0),
+				Open:   candles[i].Open,
+				High:   candles[i].High,
+				Low:    candles[i].Low,
+				Close:  candles[i].Close,
+				Volume: candles[i].Volume,
+			})
+		}
+	case asset.CoinMarginedFutures, asset.Futures:
+		candles, err := by.GetFuturesKlineData(ctx, pair, by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start)
+		if err != nil {
+			return kline.Item{}, err
+		}
+
+		for i := range candles {
+			for j := range ret.Candles {
+				if ret.Candles[j].Time.Equal(time.Unix(candles[i].OpenTime, 0)) {
+					continue
+				}
+			}
+			ret.Candles = append(ret.Candles, kline.Candle{
+				Time:   time.Unix(candles[i].OpenTime, 0),
+				Open:   candles[i].Open,
+				High:   candles[i].High,
+				Low:    candles[i].Low,
+				Close:  candles[i].Close,
+				Volume: candles[i].Volume,
+			})
+		}
+	case asset.USDTMarginedFutures:
+		candles, err := by.GetUSDTFuturesKlineData(ctx, pair, by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start)
+		if err != nil {
+			return kline.Item{}, err
+		}
+
+		for i := range candles {
+			for j := range ret.Candles {
+				if ret.Candles[j].Time.Equal(time.Unix(candles[i].OpenTime, 0)) {
+					continue
+				}
+			}
+			ret.Candles = append(ret.Candles, kline.Candle{
+				Time:   time.Unix(candles[i].OpenTime, 0),
+				Open:   candles[i].Open,
+				High:   candles[i].High,
+				Low:    candles[i].Low,
+				Close:  candles[i].Close,
+				Volume: candles[i].Volume,
+			})
+		}
+
+	default:
+		return kline.Item{}, fmt.Errorf("assetType not supported: %v", a)
+	}
+
+	dates.SetHasDataFromCandles(ret.Candles)
+	summary := dates.DataSummary(false)
+	if len(summary) > 0 {
+		log.Warnf(log.ExchangeSys, "%v - %v", by.Name, summary)
+	}
+	ret.RemoveDuplicates()
+	ret.RemoveOutsideRange(start, end)
+	ret.SortCandlesByTimestamp(false)
+	return ret, nil
+}
+
+func (b *Bybit) extractCurrencyPair(symbol string, item asset.Item) (currency.Pair, error) {
+	if len(symbol) == 6 {
+		return currency.NewPairFromString(symbol)
+	}
+	pairs, err := b.CurrencyPairs.GetPairs(item, true)
+	if err != nil {
+		return currency.Pair{}, err
+	}
+	pFmt, err := b.GetPairFormat(item, false)
+	if err != nil {
+		return currency.Pair{}, err
+	}
+	return currency.MatchPairsWithNoDelimiter(symbol, pairs, pFmt)
+}
```

Comment:
This should be faster: Please go through all websockets and change over to this method. I noticed one in spot orders handling. 
```go
                        enabled, err := by.GetEnabledPairs(item)
			if err != nil {
				return currency.Pair{}. err
			}

			return enabled.DeriveFrom(data.Symbol)

```